types/ts4.1/index.d.ts

```diff
@@ -122,22 +125,18 @@ export const Switch: FunctionComponent<SwitchProps>;
  * Components: <Router />
  */
 
-export interface RouterProps {
-  hook: BaseLocationHook;
-  base: Path;
-  matcher: MatcherFn;
-}
-export const Router: FunctionComponent<
-  Partial<RouterProps> & {
-    children: ReactNode;
-  }
->;
+export type RouterProps = Partial<RouterOptions> & {
+  parent?: RouterObject;
+  children: ReactNode;
+};
```

Comment:
@molefrog `RouterOptions` is already `Partial` and includes `parent` now. So you can just do the same thing as for TS 3.9:

```typescript
export type RouterProps = RouterOptions & {
  children: ReactNode;
};
```

You could even move the definition of `RouterProps` into the shared file since it should be exactly the same as in 3.9 if I'm not mistaken.

Reply:
You are absolutely right. I must have overlooked that when I was refactoring the options. As for the shared file, yes indeed that would be nice, supporting duplicating types becomes a burden. I have some plans on refactoring that and providing a more elegant support for 3.9.

Reply:
@molefrog `parent` can also be removed since that is already included in `RouterOptions`. Otherwise, looks good to me!